actionview/lib/action_view/template.rb

```diff
@@ -145,7 +145,17 @@ def render(view, locals, buffer=nil, &block)
         view.send(method_name, locals, buffer, &block)
       end
     rescue => e
-      handle_render_error(view, e)
+      if e.is_a?(Template::Error)
+        e.sub_template_of(self)
+        raise e
+      else
+        template = self
+        unless template.source
+          template = refresh(view)
+          template.encode!
+        end
+        raise Template::Error.new(template, e.message, e.backtrace)
+      end
```

Comment:
I think this is the only downside to this PR. We lose the flexibility to assign `#cause=`, but we rarely need it. Worthy trade.